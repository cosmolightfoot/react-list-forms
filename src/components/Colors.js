import React from 'react';
import Color from './Color.js';
import { hexFromRGB } from '../services/rgb.js';

import PropTypes from 'prop-types';

export default function Colors({ colors }) {
  const colorList = colors.map(({ name, rgb }) => {
    const hex = hexFromRGB(rgb);
    return (
      <li key={`${name}-${hex}-${rgb}`} style={{ textDecoration: 'none' }}>
        <Color name={name} hex={hex} rgb={rgb} /> 
      </li>
    );
  });

  return (
    <ul>
      {colorList}
    </ul>
  );
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};
