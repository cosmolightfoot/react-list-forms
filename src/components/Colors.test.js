import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors.js';

describe('Colors Component tests', () => {
  it('renders the colors', () => {
    const colors = [
      { name: 'Red', hex: '#FF0000' },
      { name: 'Green', hex: '#00FF00' },
      { name: 'Blue', hex: '#0000FF' }
    ];
    const wrapper = shallow(<Colors colors={colors} />);
    expect(wrapper).toMatchSnapshot();
  });
});
