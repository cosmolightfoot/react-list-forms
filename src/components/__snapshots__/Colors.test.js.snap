// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors Component tests renders the colors 1`] = `
<ul>
  <Color
    hex="#FF0000"
    key="Red+#FF0000"
    name="Red"
  />
  <Color
    hex="#00FF00"
    key="Green+#00FF00"
    name="Green"
  />
  <Color
    hex="#0000FF"
    key="Blue+#0000FF"
    name="Blue"
  />
</ul>
`;
